[azure-sdk-for-cpp] Storage Sept Release

azure-sdk opened the pull request:
Update vcpkg ports for Azure SDK release. This release may contain multiple ports.

azure-sdk committed:
[azure-storage-blobs-cpp] Update to 12.2.0
## 12.2.0 (2021-09-08)

### Breaking Changes

- `StartsOn` and `ExpiresOn` in `SignedIdentifier` were changed to nullable.

### Bugs Fixed

- Fixed a bug where prefix cannot contain `&` when listing blobs.

### Other Changes

- Create less threads if there isn't too much data to transfer.

Jinming-Hu reviewed (changes requested):
wait until `azure-storage-common-cpp` is updated

azure-sdk committed:
[azure-storage-files-datalake-cpp] Update to 12.2.0
## 12.2.0 (2021-09-08)

### Breaking Changes

- `StartsOn` and `ExpiresOn` in `SignedIdentifier` were changed to nullable.

### Bugs Fixed

- Fixed a bug where prefix cannot contain `&` when listing files.

### Other Changes

- Create less threads if there isn't too much data to transfer.

azure-sdk committed:
[azure-storage-files-shares-cpp] Update to 12.2.0
## 12.2.0 (2021-09-08)

### Breaking Changes

- `AccessPolicy::StartsOn` and `AccessPolicy::ExpiresOn` are now nullable values.

### Bugs Fixed

- Fixed a bug where prefix cannot contain `&` when listing files.

### Other Changes

- Create less threads if there isn't too much data to transfer.

azure-sdk committed:
[azure-storage-queues-cpp] Update to 12.0.0-beta.1
## 12.0.0-beta.1 (2021-09-08)

### New Features

- Added support for Queue features:
  - QueueServiceClient::ListQueues
  - QueueServiceClient::SetProperties
  - QueueServiceClient::GetProperties
  - QueueServiceClient::GetStatistics
  - QueueServiceClient::CreateQueue
  - QueueServiceClient::DeleteQueue
  - QueueClient::Create
  - QueueClient::Delete
  - QueueClient::GetProperties
  - QueueClient::SetMetadata
  - QueueClient::GetAccessPolicy
  - QueueClient::SetAccessPolicy
  - QueueClient::EnqueueMessage
  - QueueClient::ReceiveMessages
  - QueueClient::PeekMessages
  - QueueClient::UpdateMessage
  - QueueClient::DeleteMessage
  - QueueClient::ClearMessages
- Added support for queue SAS.

azure-sdk committed:
[azure-storage-common-cpp] Update to 12.2.0
## 12.2.0 (2021-09-08)

### Features Added

- Used new xml library on Windows, dropped dependency for libxml2.

### Bugs Fixed

- Fixed a bug that may cause crash when parsing XML.

BillyONeal commented:
Thanks for the update!
